.travis.yml

```
@@ -0,0 +1,148 @@
+    #- python: "3.7"
+    #- python: "3.8-dev"
+    - python: "nightly"
+    # PyPy versions
+    ## Travis Xenial doesn't support these versions
+    #- python: "pypy2.7"
+    - python: "pypy3.5"
+
+  ## Uncomment to allow the build to report success (with non-required
+  ## sub-builds continuing to run) if all required sub-builds have
+  ## succeeded.  This is somewhat buggy currently: it can cause
+  ## duplicate notifications and prematurely report success if a
+  ## single sub-build has succeeded.  See
+  ## https://github.com/travis-ci/travis-ci/issues/1696
+  #fast_finish: true
+
+  ## These builds fail in Travis at the moment
+  allow_failures:
+    - env: TOR_VERSION="0.3.5"
+    - env: TOR_VERSION="nightly"
+
+## We don't need sudo. (The "apt:" stanza after this allows us to not need
+## sudo; otherwise, we would need it for getting dependencies.)
+sudo: false
+
+## (Linux only) Use the latest Linux image (Ubuntu Xenial)
+dist: xenial
+
+## (OSX only) Use the default OSX image
+## See https://docs.travis-ci.com/user/reference/osx#os-x-version
+## Default is Xcode 9.4 on macOS 10.13 as of August 2018
+#osx_image: xcode9.4
+
+## This env var isn't used by the build, but it's useful for debugging
+## It should be kept in sync with the Linux tor version below
+env:
+  - TOR_VERSION="stable"
+
+## Download our dependencies
+addons:
+  ## (Linux only)
+  apt:
+    sources:
+      ## Xenial has Tor 0.2.4, so we need a newer version of Tor from the
+      ## torproject repositories, and the torproject key from the keyservers
+      - sourceline: 'deb https://deb.torproject.org/torproject.org xenial main'
+        key_url: 'https://deb.torproject.org/torproject.org/A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89.asc'
+    packages:
+      - tor
+  ## (macOS only)
+  homebrew:
+    packages:
+      - tor
+
+install:
+  ## Chutney has no dependencies, apart from tor
+  ## List installed package versions
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then dpkg-query --show; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew list --versions; fi
+  - python --version
+  - tor --version
+
+script:
+  - tools/test-network.sh
+
+after_failure:
+  ## List the contents of net/nodes
+  - ls -lR net/nodes/
+  ## Dump the important directory documents
+  #- cat net/nodes/000a*/cached-certs
+  - cat net/nodes/000a*/cached-consensus
+  - cat net/nodes/000a*/cached-descriptors*
+  #- cat net/nodes/000a*/cached-extrainfo*
+  - cat net/nodes/000a*/cached-microdesc-consensus
+  - cat net/nodes/000a*/cached-microdescs*
+  #- cat net/nodes/000a*/key-pinning-journal
+  #- cat net/nodes/000a*/router-stability
+  #- cat net/nodes/00*a*/sr-state
+  #- cat net/nodes/00*a*/state
+  - for f in net/nodes/00*a*/v3-status-votes ; do echo "$f"; cat "$f"; done
+  - for f in net/nodes/00*a*/unparseable-descs/* ; do echo "$f"; cat "$f"; done
+  ## Re-run chutney in debug mode
+  ## TODO: improve debug mode
+  - tools/test-network.sh --debug && echo "chutney debug run succeeded"
```

Oh, we're already doing a cheap and nasty version of test-network-forgiving!

Let's disable this and see how it goes?

```suggestion
  #- tools/test-network.sh --debug && echo "chutney debug run succeeded"
```

---

Hmm, no, we fail if the first chutney run fails. But then we do a re-run with debug output.